src/clients/common/notificationBanner.js

```diff
@@ -0,0 +1,249 @@
+      break;
+    default:
+      console.log('NotificationBanner argument type should be string or object');
+      return;
+  }
+
+  const text = notificationBannerElement.querySelector(`.notificationbanner-text`).innerHTML;
+  if (text === 'undefined' || text === '') {
+    console.log('NotificationBanner is missing text and is not displayed!');
+    return;
+  }
+
+  addNotificationBanner();
+  displayNotificationBanner();
+  clearNotificationBanners();
+
+
+  //************************************************************************************** */
+  // Helper functions for createNotificationBanner:
+
+  function getNewNotificationBanner(html) {
+    const notificationBannerDocument = getNotificationBannerDocument();
+    const notificationBannerTemplate = notificationBannerDocument.getElementById('notificationBannerTemplate');
+    const notificationBannerFragment = notificationBannerTemplate.content.cloneNode(true);
+    const notificationBannerElement = notificationBannerFragment.getElementById('notificationBanner');
+    return notificationBannerElement;
+
+    function getNotificationBannerDocument() {
+      const parser = new DOMParser();
+      const notificationBannerDocument = parser.parseFromString(html, 'text/html');
+      return notificationBannerDocument;
+    }
+
+  }
+
+  function checkNotificationBannerType() {
+
+    if (typeof notificationBannerContent === 'string' || notificationBannerContent instanceof String) {
+      return 'string';
+    }
+
+    if (typeof notificationBannerContent === 'object' && Object.keys(notificationBannerContent).length !== 0 && Object.getPrototypeOf(notificationBannerContent) === Object.prototype) {
+      return 'object';
+    }
+  }
+
+  function createDefaultNotificationBanner(notificationBannerString) {
+    createCustomNotificationBanner({text: notificationBannerString});
+  }
+
+  function createCustomNotificationBanner(notificationBannerObject) {
+    const {style, text, linkButton} = notificationBannerObject;
+
+    setNotificationBannerStyle();
+    addTextToNotificationBanner(text);
+    addLinkButton();
+    addCloseButton(notificationBannerObject);
+
+    function setNotificationBannerStyle() {
+
+      // style 'info' is the default status message
+      // notificationBanner has blue background with info icon
+      let backgroundColor = 'var(--color-blue-60)';
+      let iconColor = 'var(--color-blue-100)';
+      let icon = 'error_outline';
+
+
+      if (style === 'success') {
+        backgroundColor = 'var(--color-green-80)';
+        iconColor = 'var(--color-green-100)';
+        icon = 'check_circle_outline';
+      }
+
+      if (style === 'alert') {
+        backgroundColor = 'var(--color-yellow-80)';
+        iconColor = 'var(--color-blue-100)';
+        icon = 'warning_amber';
+      }
+
+      if (style === 'error') {
+        backgroundColor = 'var(--color-red-80)';
+        iconColor = 'var(--color-red-100)';
+        icon = 'report_gmailerrorred';
+      }
+
```

Comment:
create helper function that return object containing all 3 colors and takes style as param. If param is undefined return default style

Reply:
Now using helper function and using returned object in banner/banner_static and dialog.